autosklearn/automl.py

```diff
@@ -1631,6 +1633,94 @@ def _load_best_individual_model(self):
         )
         return ensemble
 
+    def _load_pareto_front(self) -> Sequence[VotingClassifier | VotingRegressor]:
+        if len(self._metrics) <= 1:
+            raise ValueError("Pareto front is only available for two or more metrics.")
+
+        if self._ensemble_size > 0:
+            self.ensemble_ = self._backend.load_ensemble(self._seed)
+        else:
+            self.ensemble_ = None
+
+        # If no ensemble is loaded we cannot do anything
+        if not self.ensemble_:
+            raise ValueError(
+                "Pareto front can only be accessed if an ensemble is available."
+            )
+
+        elif not isinstance(self.ensemble_, MultiObjectiveEnsembleWrapper):
+            raise ValueError(
+                "Can only retrieve a Pareto front from a multi-objective ensemble "
+                "builder."
+            )
+
+        else:
+            ensembles = []
+            for ensemble in self.ensemble_.get_pareto_front():
+                identifiers = ensemble.get_selected_model_identifiers()
+                weights = {
+                    identifier: weight
+                    for identifier, weight in ensemble.get_identifiers_with_weights()
+                }
+
+                if self._task in CLASSIFICATION_TASKS:
+                    voter = VotingClassifier(
+                        estimators=None,
+                        voting="soft",
+                    )
+                else:
+                    voter = VotingRegressor(estimators=None)
+
+                if self._resampling_strategy in ("cv", "cv-iterative-fit"):
+                    models = self._backend.load_cv_models_by_identifiers(identifiers)
+                else:
+                    models = self._backend.load_models_by_identifiers(identifiers)
+
+                if len(models) == 0:
+                    raise ValueError("No models fitted!")
+
+                weight_vector = []
+                estimators = []
+                for identifier in identifiers:
+                    weight_vector.append(weights[identifier])
+                    estimators.append(models[identifier])
+
+                voter.estimators = estimators
+                voter.estimators_ = estimators
+                voter.weights = weight_vector
+
+                if self._task in CLASSIFICATION_TASKS:
+                    # Scikit-learn would raise a shape error here which we
+                    # have to work around...
+
+                    def inverse_transform(self, y):
+                        if len(y.shape) == 1:
+                            y = y.reshape((-1, 1))
+                            reshaped = True
+                        else:
+                            reshaped = False
+                        y = self.old_inverse_transform(y)
+                        # Not sure why scikit-learn transforms to type object...
```

Comment:
So scikit learn will for some reason change the type of `y` even if it's already valid?

Reply:
I don't understand this myself yet. It does an "upcast" if "ignored" values are found: https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/preprocessing/_encoders.py#L1030 but I don't understand where they come from. I'll now try to figure that out.

Reply:
Alright, this can be circumvented by passing the argument `handle_unknown="error"` to the label encoder: https://github.com/automl-private/auto-sklearn/pull/3/commits/b23e912827d90417bbf74883e0e43841f498b520#diff-2da77436544f4bb378cca56ab2cb993c13af10e102b7ba694d2474ef4192446fR174

Using `"error"` also will not hurt because the label encoder is fit on the training data, which is all data Auto-sklearn will ever get (for `y`).